File: src/dere_shared/paths.py
"""Shared filesystem helpers for dere projects."""

from __future__ import annotations

import os
import platform
from pathlib import Path


def get_config_dir() -> Path:
    """Get platform-specific config directory."""
    match platform.system():
        case "Windows":
            return Path(os.getenv("LOCALAPPDATA", "")) / "dere"
        case "Darwin":
            return Path.home() / "Library" / "Application Support" / "dere"
        case _:
            return Path.home() / ".config" / "dere"


def get_data_dir() -> Path:
    """Get platform-specific data directory."""
    match platform.system():
        case "Windows":
            return Path(os.getenv("LOCALAPPDATA", "")) / "dere"
        case "Darwin":
            return Path.home() / "Library" / "Application Support" / "dere"
        case _:
            return Path.home() / ".local" / "share" / "dere"

File: packages/shared-runtime/src/paths.ts
import { homedir } from "node:os";
import { join, dirname } from "node:path";

import { getConfigPath } from "@dere/shared-config";

export function getConfigDir(): string {
  return dirname(getConfigPath());
}

export function getDataDir(): string {
  if (process.platform === "darwin") {
    return join(homedir(), "Library", "Application Support", "dere");
  }
  if (process.platform === "win32") {
    return join(process.env.LOCALAPPDATA ?? homedir(), "dere");
  }
  return join(homedir(), ".local", "share", "dere");
}
